src/pages/auth/signin.tsx

```diff
@@ -47,60 +92,80 @@ const SignIn: NextPage = () => {
       });
   });
 
-  const onSubmit = useCallback<Parameters<typeof handleSubmit>[0]>(
-    async ({ email }) => {
-      const callbackUrl = router.query.callbackUrl as string;
-      setIsLoading(true);
+  const [selectedRole, setSelectedRole] = useState<UserRole | null>(null);
 
-      const { ok } = (await signIn<'email'>('email', {
-        email,
-        callbackUrl,
-      })) as SignInResponse;
-      if (!ok) {
-        setIsLoading(false);
+  useOnce(
+    () => {
+      if (
+        queryUser === UserRole.Applicant ||
+        queryUser === UserRole.Requestor
+      ) {
+        setSelectedRole(queryUser);
+        return;
       }
+      setSelectedRole(UserRole.Applicant);
```

Comment:
```suggestion
      if (
        queryUser === UserRole.Requestor
      ) {
        setSelectedRole(UserRole.Requestor);
      } else {
        setSelectedRole(UserRole.Applicant);
      }
```
nothing necessary, just a bit easier readable imo

Reply:
I wanted it to be explicit so no one could set it's role as admin, but that way is cleaner 😌